### PR TITLE
CRM-21027 Fix next recurring payment not accurately calculated

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
This pull request fixes a bug that entered the code base in May, causing payment processors that rely on CiviCRM updating the next_sched_contribution_date for recurring to not get it updated.

Before
----------------------------------------
When payments are completed using the completetransaction api with a receive_date passed in with a time component the date for the next scheduled contribution is not updated. This is a critical error to sites affected as customers will be recharged, but mitigated by the fact that relatively few use the combination of components (in my case DPS / Omnipay + recurring payments)

After
----------------------------------------
After processing a payment the next_sched_contribution_date is correctly updated

Technical Details
----------------------------------------
Rather than comparing effectiveDate with today, the date part of effectiveDate ('Y-m-d') is compared with today

Comments
----------------------------------------
I was able to replicate this in a unit test before fixing it & committing the test

---

 * [CRM-21027: Next recurring payment not accurately calculated when effective date passed in](https://issues.civicrm.org/jira/browse/CRM-21027)

### CODE REVIEW

File: tests/phpunit/CRM/Contribute/BAO/ContributionRecurTest.php [hunk 1]
```diff
@@ -99,15 +99,15 @@ public function testCancelRecur() {
    */
   public function testSupportFinancialTypeChange() {
     $contributionRecur = $this->callAPISuccess('contribution_recur', 'create', $this->_params);
-    $contribution = $this->callAPISuccess('contribution', 'create', array(
+    $this->callAPISuccess('contribution', 'create', array(
```
Comment (eileenmcnaughton) [inline tests/phpunit/CRM/Contribute/BAO/ContributionRecurTest.php]: these changes are unrelated - I noticed these minor errors while looking in the wrong file

**Comment by petednz** [comment]
In 'before' it says "as customers will be recharged". Is there a 'not' missing from that statement

**Comment by eileenmcnaughton** [comment]
@petednz no - there isn't! I narrowly avoiding multi-charging yes customers due there being another bug in the Omnipay extensions meaning they weren't charged at all for 4 days

**Comment by monishdeb** [comment]
The fix makes sense. Tested, working fine. Also the added use-case in unit test is accurate. Merging now.
